package workrave.networking;

message Domain
{
    extensions 100 to max;
}


message Activity
{
    extend Domain
    {
        required Domain type = 100;
    }
    
    required bool active = 1;
}

message Break
{
    extend Domain
    {
        required Domain type = 101;
    }

    enum BreakEvent
    {
        // No event
        BREAK_EVENT_NONE = 1;

        // The user choose to postpone a break
        BREAK_EVENT_USER_POSTPONE = 2;

        // The user choose to skip a break
        BREAK_EVENT_USER_SKIP = 3;

        // The user chooses to start the break manually
        BREAK_EVENT_FORCE_BREAK = 4;
    }


    required int32 break_id = 1;

    required BreakEvent break_event = 2;

    required int32 break_hint = 3;
}

message Configuration
{
    extend Domain
    {
        required Domain type = 102;
    }

    enum Reason
    {
      INITIAL = 1;
      USER = 2;
      RESOLVED = 3;
    }
    
    message Setting
    {
        required string key = 1;
        required string value = 2;
    }

    repeated Setting changes = 1;
    required Reason reason = 2;
}

message OperationMode
{
    extend Domain
    {
        required Domain type = 103;
    }

    enum Mode
    {
      //! Operation mode change to suspended
      OPERATION_MODE_SUSPENDED = 1;

      //! Operation mode change to quiet
      OPERATION_MODE_QUIET = 2;

      //! Operation mode change to normal
      OPERATION_MODE_NORMAL = 3;
    };

    required Mode mode = 1;
}

message UsageMode
{
    extend Domain
    {
        required Domain type = 104;
    }

    enum Mode
    {
        USAGE_MODE_NORMAL = 1;
        USAGE_MODE_READING = 2;
    };
    
    required Mode mode = 1;
}

message Timer
{
    extend Domain
    {
        required Domain type = 105;
    }

    repeated int32 idle_times = 1;
    repeated int32 active_times = 2;
}